### ISSUE TITLE
service.patch leads to 404 if query contains the patched field

### ISSUE THREAD

**Issue by toovy** [feathersjs-ecosystem/feathers-mongoose#345]
Hi,

this is a cross-post from feathers-hooks-commons, as I'm not sure, which component is responsible for this issue. I'l summarise the relevant parts here, for more details please also see https://github.com/feathers-plus/feathers-hooks-common/issues/532.

**Summary**

In the code at

https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/master/lib/service.js#L278

mongoose is used to patch a record, using the the filtered `params.query`, and in L286 this query is more or less used to find the same record again after the patch. In the case of softDelete2 the query contains `{deletedAt: -1}` that should be probably used during the patch call as query parameter. In L278 the query still contains `{deletedAt: -1}`, but as data has set deletedAt to another value in between the record is not found, leading to a 404.

This is probably true for all patch calls that use a value A as query and modify this value A during the patch call.

Any thoughts?

**Comment by Director99** [comment]
@toovy I am also a problem like you, and my error after patch 'NotFound: No record found for id '5d8a001bae64e4e9a8c74aa8'

**Comment by daffl** [comment]
If someone can provide a small breaking example that can be included in a test it would be much easier to start a fix.

**Comment by toovy** [comment]
@daffl I've provided an example at https://github.com/triggercode/feathers-mongoose-patch-issue.

The example has two branches, master and feathers-mongoose-bug. In both the services tests are used to reproduce the bug.

To setup the master branch a `npm i` is enough, then `npm run test` shows that the issues service that is based on the memory adapter works with softDelete2. The record is returned successfully on remove.

The setup of the feathers-mongoose-bug is a bit more complex as we need a mongodb. So I've included everything needed, assuming that docker is installed:

```
$ npm i
$ docker-compose up -d
$ npm run init-db
$ npm test
```

I noticed that there might be an timinig issue as well. Looking at my await statements everything seems to be correct. Mostly all tests fail, randomly the first one passes and returns a record.

Thanks for looking into it!

**Comment by arfanliaqat** [comment]
Greetings... Any update on this?

**Comment by jnardone** [comment]
Related issue: https://github.com/feathersjs-ecosystem/feathers-mongoose/issues/321

**Comment by rlunden** [comment]
This seems to happen with feathers-sequelize as well

**Comment by arfanliaqat** [comment]
this is fixed in #362
